### PR TITLE
CNCF Federated Governance Rebase

### PR DESCRIPTION
**Pull request by asmacdo**
This is the start of a rebase of our Governance model based on the [CNCF template for Federated Projects](https://github.com/cncf/project-template/blob/main/GOVERNANCE-subprojects.md). 

Feedback and discussion is encouraged, including open office hours each Wednesday at 3pm Eastern. 
Meeting link: https://zoom.us/j/8415370125
Password: 77777

 - [X] ~This PR will also include the Contributor Ladder, both should be merged together.~

### CODE REVIEW

**Comment by asmacdo** [comment]
This PR will also include the Contributor Ladder, both should be merged together.

File: GOVERNANCE.md
Comment (kevinrizza) [inline GOVERNANCE.md:48]: nit: grammar? Maybe `and managing the lifecycle of Kubernetes Operators`?

File: GOVERNANCE.md
Comment (awgreene) [inline GOVERNANCE.md:105]: Needs info.

File: contributor-ladder.md
Comment (awgreene) [inline contributor-ladder.md:69]: Nit: Newline.

File: contributor-ladder.md
Comment (awgreene) [inline contributor-ladder.md:136]:
```suggestion
contributors to step in. Involuntary removal or demotion is handled
```

**Review by awgreene — CHANGES_REQUESTED** [review]
Great work! A few nits.

File: GOVERNANCE.md
Comment (joelanford) [inline GOVERNANCE.md:162]: Should we explicitly call out that the Steering Committee can delegate administration of project infrastructure (or other Steering Committee duties)?

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:229]: A majority of the Maintainers and Approvers must approve.

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:226]: , all of the following must occur

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:224]: 4. The vote must remain open for a minimum of 7 days, the Steering Committee may extend the period.

Comment (joelanford) [inline GOVERNANCE.md:224]: If a subproject is very conservative with maintainer/approver status, they will have less influence than another similarly sized subproject that gives every contributor maintainer/approver status.

It seems like our voting rules are setting up incentives for subprojects to "race to the bottom" in terms of doling out approver/maintainer to swing votes in their favor.

Is there a way to structure general votes to enable subprojects to have appropriate influence in elections regardless of their internal contributor ladder behavior?

Perhaps we just allow all org members to vote regardless of their position on a contributor ladder? And we just require $some quorum of org members?

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:265]: + Subproject has at least 3 Approvers (Experimental Subprojects must have at least 1)

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:270]: How many Steering Committee votes are required to accept?

Comment (asmacdo) [reply GOVERNANCE.md:270]: 2/3 for graduated 1/2 for experimental

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:284]: s/majority/ 2\/3/

File: GOVERNANCE.md
Comment (jberkhahn) [inline GOVERNANCE.md:151]: We decided to limit CNCF Maintainer status to just the Steering committee, right? Should have a blurb in here about representing OF to CNCF and related things like Kubecon

File: GOVERNANCE.md
Comment (asmacdo) [reply GOVERNANCE.md:224]: We discussed this one with TAG Contributor Strategy
 https://cloud-native.slack.com/archives/CT6CWS1JN/p1660759198667419?thread_ts=1660759198.667419&cid=CT6CWS1JN


"""
@jberkus to be clear, what you are suggesting is:
l
Governance Amendments are approved via a 2/3 majority of the Steering Committee. 

each subproject will have at least 1 voting rep based on steering committee membership
"""

File: GOVERNANCE.md
Comment (jmrodri) [inline GOVERNANCE.md:5]:
```suggestion
which focuses on some aspect of the goals of the umbrella project. Our
```

File: GOVERNANCE.md
Comment (jberkus) [reply GOVERNANCE.md:224]: Yes, although the rep count is covered elsewhere in the document.

File: GOVERNANCE.md
Comment (asmacdo) [reply GOVERNANCE.md:150]: added in latest commit

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:69]: maintain tooling of catalogs

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:140]: This feels like a recipe to make it impossible to ever change governance.  Consider that the #1 reason to change written governance is that the prior rules have become inoperable, usuaully due to contributor departures.  Suggest instead:

```suggestion
If a subproject wishes to update or supplement governance
documents, this will require a 2/3 majority of project Approvers
or 2/3 majority approval from the Operator Framework steering
committee.
```

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:71]: Managing the lifecycle of Operators

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:73]: The of community develops best practices, fosters collaboration, and improves tool interoperability

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:230]: There's issues with quorums; will discuss during the meeting.

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:140]: lets remove the 2/3 of approvers

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:42]: +consensus seeking

File: GOVERNANCE.md
Comment (joelanford) [inline GOVERNANCE.md:148]:
```suggestion
* `7 - number_of_subprojects` OR 3 general "community representatives", whichever is greater.
```

If number of subprojects is >= 7, we still want community reps, so the minimum number of reps is 3.
Also, we don't need to worry about even/odd concerns because:
- we don't have very many simple majority votes
- even if we do, votes typically don't garner 100% turnout (i.e. 10 of 11 votes might vote and split 5/5).

Comment (joelanford) [reply GOVERNANCE.md:148]: Also Josh suggested we add a table to show some examples.

File: GOVERNANCE.md
Comment (jberkhahn) [inline GOVERNANCE.md:128]: Add a blurb here explicitly saying if the subproject only puts forward a single nominee, then don't bother holding an election.

File: GOVERNANCE.md
Comment (jberkus) [inline GOVERNANCE.md:272]: Rather than having quorums, my suggestion is that you simply require all SC votes to be as a % of all existing Steering seats.  That is, if there's 7 commitee members then 4 are required for any majority vote and 5 for any 2/3 vote, regardless of how that vote is taken.  Quorums, especially minority quorums, must have rules around notice periods etc. that can be annoying to administer.

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:270]:
```suggestion
must hold an internal vote of Approvers to join
```

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:295]:
```suggestion
```
Lets leave this to the discretion of the steering committee

File: contributor-ladder.md
Comment (asmacdo) [inline contributor-ladder.md:33]: this one doesnt really mean anything, lets remove

File: contributor-ladder.md
Comment (asmacdo) [inline contributor-ladder.md:85]: subproject

File: contributor-ladder.md
Comment (asmacdo) [inline contributor-ladder.md:99]: same subproject

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:219]: include ladder

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:255]: when a new subproject is added (and they gain a steering committee seat) This seat will be additive, (no existing community chairs will be removed, even if there are fewer at large seats next time)

File: contributor-ladder.md
Comment (jberkhahn) [inline contributor-ladder.md:126]: make this link to the member section

File: GOVERNANCE.md
Comment (csviri) [inline GOVERNANCE.md:211]: This is not really clear to me, so the current 7 chairs will be steering committee for the next year?

Comment (asmacdo) [reply GOVERNANCE.md:211]: yes, thats correct. FWIW, if we adopt a new Subproject, they will also get a rep, so we would have 8 or more

File: GOVERNANCE.md
Comment (asmacdo) [inline GOVERNANCE.md:199]: cue: spellcheck

File: GOVERNANCE.md
Comment (asmacdo) [reply GOVERNANCE.md:240]: We discussed this in the office hour-- this is vague, and up to the Steering committee to determine.

File: GOVERNANCE.md
Comment (asmacdo) [reply GOVERNANCE.md:41]: when discussed in the office hour, it was determined that consensus seeking is part of being inclusive

File: GOVERNANCE.md
Comment (asmacdo) [reply GOVERNANCE.md]: I removed the algebra, should be clearer now.

File: GOVERNANCE.md
Comment (asmacdo) [reply GOVERNANCE.md]: discretion of the steering committee is implied since they will be voting.

**Comment by asmacdo** [comment]
+1

**Comment by everettraven** [comment]
+1

**Comment by grokspawn** [comment]
+1

**Comment by oceanc80** [comment]
+1

**Comment by jberkhahn** [comment]
+1

**Comment by jmrodri** [comment]
+1

**Comment by dmesser** [comment]
+1

**Comment by fabianvf** [comment]
+1

**Comment by perdasilva** [comment]
+1

**Comment by anik120** [comment]
+1

**Comment by dinhxuanvu** [comment]
+1

**Comment by akihikokuroda** [comment]
+1

**Comment by ankitathomas** [comment]
+1

**Comment by dmueller2001** [comment]
+1

**Comment by timflannagan** [comment]
+1

**Comment by varshaprasad96** [comment]
+1

**Comment by awgreene** [comment]
+1

**Comment by gallettilance** [comment]
+1

**Comment by ecordell** [comment]
+1

**Review by awgreene — APPROVED** [review]
/approve

**Comment by tlwu2013** [comment]
+1

**Pull request merged**
